### DISCUSSION TITLE
Bulk free energy specification & ADAllenCahn

### DISCUSSION THREAD

**Discussion by Leni-Yeo** [idaholab/moose#17093] [Q&A Modules: Phase field] [answered]
Good morning, my name is Leni. I am a student and relatively new in using MOOSE / Phase Field and have few questions for anyone able to help.

1) I am trying to do a simulation for polycrystal grain growth. I saw the example that applies the polycrystalvariables and polycrystalkernel actions. I modified the polycrystalkernel action to include the following kernels: ADAllenCahn, ADTimeDerivative, and ADACInterfaceKobayashi2 for a basic simulation. My trouble comes in with the bulk free energy [gr^2 * (gr - 1)^2)] that I need to specify as a function material. I tried to write it as a derivative parsed material directly under the Materials block but since we have many order parameters set up with the polycrystalvariables action (gr0, gr1, gr2,...), the software does not recognizes the variable in the equation. I am wondering if there is a specific format in writing the equation in the input file to refer to all the parameters or if there is any specific kernel that already produces this outcome that I may have missed.

2) The bulk free energy that I am using only depends on the order parameter [gr^2 * (gr - 1)^2)] which I can use the ADAllenCahn kernel to implement. But I was wondering if there is a version of the ADAllenCahn kernel that takes into account bulk free energy equations with parameters/arguments other than the order parameter. I know the AllenCahn Kernel does that but I am asking about one that has automatic differentiation applied in . 

I attached my input file and the kernel action I am using to set up the kernels.

[files.zip](https://github.com/idaholab/moose/files/6024250/files.zip)

**Comment by SudiptaBiswas** [comment, marked as answer]
Symbolic representation in the input file doesn't work. You have to provide the bulk energy for all the order parameters. For isotropic grain growth, you can use the `GrainGorwthAction`  with `use_automatic_differentiation = true` to activate the AD objects.

**Reply by Leni-Yeo** [reply]
I see, thank you for the help.
